src/main.ts

```diff
@@ -70,12 +70,19 @@ async function setupBuildx(version: string, toolkit: Toolkit): Promise<void> {
  */
 export async function startBlacksmithBuilder(inputs: context.Inputs): Promise<{addr: string | null; buildId: string | null; exposeId: string}> {
   try {
-    const dockerfilePath = context.getDockerfilePath(inputs);
-    if (!dockerfilePath) {
-      throw new Error('Failed to resolve dockerfile path');
+    let dockerfilePath: string | null = null;
+    // We only use the dockerfile path to report the build to our control plane.
+    // If setup-only is true, we don't want to report the build to our control plane
+    // since we are only setting up the builder and therefore cannot expose any analytics
+    // about the build.
+    if (!inputs.setupOnly) {
+      dockerfilePath = context.getDockerfilePath(inputs);
+      if (!dockerfilePath) {
+        throw new Error('Failed to resolve dockerfile path');
+      }
```

Comment:
One small cleanup suggestion that will also cleanup the calls below.

```suggestion
    // We only use the dockerfile path to report the build to our control plane.
    // If setup-only is true, we don't want to report the build to our control plane
    // since we are only setting up the builder and therefore cannot expose any analytics
    // about the build.
    const dockerfilePath = inputs.setupOnly ? "" : context.getDockerfilePath(inputs);
    if (!inputs.setupOnly && dockerfilePath === "") {
        throw new Error('Failed to resolve dockerfile path');
    let dockerfilePath: string | null = null;
```